frontend/src/modules/entitydetails/components/Metadata.js

```diff
@@ -84,6 +90,56 @@ export default class Metadata extends React.Component<Props> {
         </Localized>;
     }
 
+    renderGroupComment(entity: Entity): React.Node {
+        if (!entity.group_comment) {
+            return null;
+        }
+
+        return <Localized id='entitydetails-Metadata--group-comment' attrs={ { title: true } }>
+            <Property title='Group Comment' className='comment'>
+                <Linkify properties={ { target: '_blank', rel: 'noopener noreferrer' } }>
+                    { entity.group_comment }
+                </Linkify>
+            </Property>
+        </Localized>;
+    }
+
+    constructor(props: Props) {
+        super(props);
+        this.state = { seeMore: false };
+    }
+
+    handleClick = () => {
+        this.setState((state) => {
+            return { seeMore: !state.seeMore };
+        });
+    };
+
+    renderResourceComment(entity: Entity): React.Node {
+        if (!entity.resource_comment) {
+            return null;
+        }
+
+        return <Localized id='entitydetails-Metadata--resource-comment' attrs={ { title: true } }>
+            <Property
+                title='Resource Comment'
+                className={ (this.state.seeMore) ? 'comment' : 'comment truncate' }
+            >
+                <Linkify properties={ { target: '_blank', rel: 'noopener noreferrer' } }>
+                    { entity.resource_comment }
+                </Linkify>
+                <Localized id='entitydetails-Metadata--see-more'>
+                    <button
+                        className={ (this.state.seeMore) ? 'hide' : '' }
```

Comment:
This is a jQueryish approach. In React world we don't even render the button if `seeMore === true`. Then you can also remove the `.hide` rule from CSS.

For the exact code pattern, see: https://github.com/mozilla/pontoon/blob/df0aafcf03336462cfd1419dd2c908e87d7f1a97/frontend/src/modules/entitydetails/components/Helpers.js#L76-L78.